examples/wakaama/lwm2mclient.c

```diff
@@ -0,0 +1,695 @@
+                fprintf(stdout, "STATE_BOOTSTRAPPING\r\n");
+                break;
+            case STATE_REGISTER_REQUIRED:
+                fprintf(stdout, "STATE_REGISTER_REQUIRED\r\n");
+                break;
+            case STATE_REGISTERING:
+                fprintf(stdout, "STATE_REGISTERING\r\n");
+                break;
+            case STATE_READY:
+                fprintf(stdout, "STATE_READY\r\n");
+                break;
+            default:
+                fprintf(stdout, "Unknown...\r\n");
+                break;
+        }
+        /*
+         * This function does two things:
+         *  - first it does the work needed by liblwm2m (eg. (re)sending some
+         * packets).
+         *  - Secondly it adjusts the timeout value (default 60s) depending on the
+         * state of the transaction
+         *    (eg. retransmission) and the time between the next operation
+         */
+        DEBUG("lwm2m_step()\n");
+        result = lwm2m_step(lwm2mH, &(tv.tv_sec));
+        fprintf(stdout, " -> State: ");
+        switch (lwm2mH->state) {
+            case STATE_INITIAL:
+                fprintf(stdout, "STATE_INITIAL\r\n");
+                break;
+            case STATE_BOOTSTRAP_REQUIRED:
+                fprintf(stdout, "STATE_BOOTSTRAP_REQUIRED\r\n");
+                break;
+            case STATE_BOOTSTRAPPING:
+                fprintf(stdout, "STATE_BOOTSTRAPPING\r\n");
+                break;
+            case STATE_REGISTER_REQUIRED:
+                fprintf(stdout, "STATE_REGISTER_REQUIRED\r\n");
+                break;
+            case STATE_REGISTERING:
+                fprintf(stdout, "STATE_REGISTERING\r\n");
+                break;
+            case STATE_READY:
+                fprintf(stdout, "STATE_READY\ren");
+                break;
+            default:
+                fprintf(stdout, "Unknown...\r\n");
+                break;
+        }
+        if (result != 0) {
+            fprintf(stderr, "lwm2m_step() failed: 0x%X\r\n", result);
+#ifdef LWM2M_BOOTSTRAP
+            if (previousState == STATE_BOOTSTRAPPING) {
+#ifdef WITH_LOGS
+                fprintf(stdout,
+                        "[BOOTSTRAP] restore security and server objects\r\n");
+#endif          /* WITH_LOGS */
+                prv_restore_objects(lwm2mH);
+                lwm2mH->state = STATE_INITIAL;
+            }
+            else
+#endif      /* LWM2M_BOOTSTRAP */
+            {
+                g_quit = 1;
+                continue;
+            }
+        }
+#ifdef LWM2M_BOOTSTRAP
+        DEBUG("update_bootstrap_info()\r\n");
+        update_bootstrap_info(&previousState, lwm2mH);
+#endif
+
+#ifdef MODULE_TINYDTLS
+handshake:
+#endif  /* MODULE_TINYDTLS */
+        DEBUG("xtimer_msg_receive_timeout64()\r\n");
+        //if (xtimer_msg_receive_timeout64(&msg, tv.tv_sec * 1000000) >= 0) {
+        pckt_rcvd_size = sock_udp_recv(&data.sock, &packet_rcvd, RCVD_SIZE, tv.tv_sec * US_PER_SEC, &remote);
```

Comment:
I'm just trying to understand this part.

Please, can you elaborate  why are you using tv.tv_sec for the timeout here? Is not more practical to wait a specific amount of time ? (like a single second).